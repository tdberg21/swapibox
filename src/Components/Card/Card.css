.card {
  border: 1px solid black;
  margin: 8px;
  padding-left: 5px;
  padding-right: 5px;
  padding-bottom: 10px;
  width: 29%;
  background-color: lightgrey;
}

.favorite-button {
  border-color: yellow;
  color: yellow;
  background-color: #333;
}

.favorite {
  background-color: chartreuse;
}

ul {
  list-style-type: none;
  padding-left: 0;
}

li {
  text-overflow: ellipsis;
  padding-right: 5px;
}