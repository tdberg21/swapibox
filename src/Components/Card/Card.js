import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';


const Card = (props) => {

  let cardToRender = Object.keys(props.info).map(item => {
    console.log(props.info[item])
    return <p> {[item]} : {props.info[item]} </p>
  });

  return (
    <div className="card">
      {cardToRender}
      <button
        onClick={(event) => props.checkForFaves(props.info.id, event)}
        className="favorite-button">
          Favorite
      </button>
    </div>
  );


  // if (props.info.species) {
  //   return (
  //     <div className="card">
  //       <h5>{props.info.name}</h5>
  //       <p>Species: {props.info.species} </p>
  //       <p>Homeworld: {props.info.homeworld}</p>
  //       <p>Homeworld population:{props.info.population}</p>
  //       <button 
  //         onClick={(event) => props.checkForFaves(props.info.id, event)}
  //         className="favorite-button">
  //         Favorite
  //       </button>
  //     </div>
  //   );
  // } 
  
  // if (props.info.terrain) {
  //   const residentsToDisplay = props.info.residents.map((resident, index) => {
  //     return (<li key={index}>{resident}</li>);
  //   });
    
  //   return (
  //     <div className="card">
  //       <h5>{props.info.name}</h5>
  //       <p>Climate: {props.info.climate} </p>
  //       <p>Terrain: {props.info.terrain}</p>
  //       <p>Population:{props.info.population}</p>
  //       <ul>Residents:{residentsToDisplay}</ul>
  //       <button 
  //         onClick={(event) => props.checkForFaves(props.info.id, event)}
  //         className="favorite-button"
  //       >
  //       Favorite
  //       </button>
  //     </div>
  //   );
  // } 
   
  // if (props.info.model) {
  //   return (
  //     <div className="card">
  //       <h5>{props.info.name}</h5>
  //       <p>Model: {props.info.model} </p>
  //       <p>Class: {props.info.class}</p>
  //       <p>Passengers:{props.info.passengers}</p>
  //       <button 
  //         onClick={(event) => props.checkForFaves(props.info.id, event)}
  //         className="favorite-button"
  //       >
  //       Favorite
  //       </button>
  //     </div>
  //   );
  // }
};

Card.propTypes = {
  name: PropTypes.string,
  species: PropTypes.string,
  homeWorld: PropTypes.string,
  homePop: PropTypes.number,
  info: PropTypes.object,
  checkForFaves: PropTypes.func
};

export default Card;
