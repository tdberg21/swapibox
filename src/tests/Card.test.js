import Card from '../Components/Card/Card.js';
import React from 'react';
import { shallow } from 'enzyme';


describe('CARD TESTS', () => {
  let wrapper;
  let mockCheckForFaves = jest.fn();
  
  it('matches the snapshot when a people are passed down', () => {
    let mockPeopleCard = { homeworld: "Tatooine", id: 1530987621630, name: "Luke Skywalker", population: "200000", species: "Human" };
    wrapper = shallow(<Card
      info={mockPeopleCard}
      key="0"
      checkForFaves={mockCheckForFaves}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot when a vehicle is passed down', () => {
    let mockPlanetCard = {
      climate: "temperate", id: 1530987442516, planet: "Alderaan", population: "2000000000", residents: ["Leia Organa", "Bail Prestor Organa", "Raymus Antilles"], terrain: "grasslands, mountains"};
    wrapper = shallow(<Card
      info={mockPlanetCard}
      key="0"
      checkForFaves={mockCheckForFaves}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot when a vehicle is passed down', () => {
    let mockVehicleCard = { class: "wheeled", id: 0, model: "Digger Crawler", name: "Sand Crawler", passengers: "30" };
    wrapper = shallow(<Card
      info={mockVehicleCard}
      key="0"
      checkForFaves={mockCheckForFaves}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke addToFaves function when a favorite button is clicked on a vehicle card', () => {
    let mockVehicleCard = { class: "wheeled", id: 0, model: "Digger Crawler", name: "Sand Crawler", passengers: "30" };
    wrapper = shallow(<Card
      info={mockVehicleCard}
      key="0"
      checkForFaves={mockCheckForFaves}
    />);

    wrapper.find('button').simulate('click');
    expect(mockCheckForFaves).toHaveBeenCalled();
  });

  it('should invoke addToFaves function when a favorite button is clicked on a planet card', () => {
    let mockPlanetCard = {
      climate: "temperate", id: 1530987442516, planet: "Alderaan", population: "2000000000", residents: ["Leia Organa", "Bail Prestor Organa", "Raymus Antilles"], terrain: "grasslands, mountains"
    };
    wrapper = shallow(<Card
      info={mockPlanetCard}
      key="0"
      checkForFaves={mockCheckForFaves}
    />);

    wrapper.find('button').simulate('click');
    expect(mockCheckForFaves).toHaveBeenCalled();
  });

  it('should invoke addToFaves function when a favorite button is clicked on a person', () => {
    let mockPeopleCard = { homeworld: "Tatooine", id: 1530987621630, name: "Luke Skywalker", population: "200000", species: "Human" };
    wrapper = shallow(<Card
      info={mockPeopleCard}
      key="0"
      checkForFaves={mockCheckForFaves}
    />);

    wrapper.find('button').simulate('click');
    expect(mockCheckForFaves).toHaveBeenCalled();
  });

});


