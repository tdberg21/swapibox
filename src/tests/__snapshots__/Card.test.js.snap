// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD TESTS matches the snapshot when a people are passed down 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    checkForFaves={[Function]}
    info={
        Object {
            "homeworld": "Tatooine",
            "id": 1530987621630,
            "name": "Luke Skywalker",
            "population": "200000",
            "species": "Human",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Luke Skywalker
</h5>,
        <p>
          Species: 
          Human
           
</p>,
        <p>
          Homeworld: 
          Tatooine
</p>,
        <p>
          Homeworld population:
          200000
</p>,
        <button
          className="favorite-button"
          onClick={[Function]}
>
          Favorite
</button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Species: ",
            "Human",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Species: ",
          "Human",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld: ",
            "Tatooine",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Homeworld: ",
          "Tatooine",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld population:",
            "200000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Homeworld population:",
          "200000",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "favorite-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Luke Skywalker
</h5>,
          <p>
            Species: 
            Human
             
</p>,
          <p>
            Homeworld: 
            Tatooine
</p>,
          <p>
            Homeworld population:
            200000
</p>,
          <button
            className="favorite-button"
            onClick={[Function]}
>
            Favorite
</button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species: ",
              "Human",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Species: ",
            "Human",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              "Tatooine",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
            "Tatooine",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld population:",
              "200000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld population:",
            "200000",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "favorite-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CARD TESTS matches the snapshot when a vehicle is passed down 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    checkForFaves={[Function]}
    info={
        Object {
            "climate": "temperate",
            "id": 1530987442516,
            "planet": "Alderaan",
            "population": "2000000000",
            "residents": Array [
              "Leia Organa",
              "Bail Prestor Organa",
              "Raymus Antilles",
            ],
            "terrain": "grasslands, mountains",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5 />,
        <p>
          Climate: 
          temperate
           
</p>,
        <p>
          Terrain: 
          grasslands, mountains
</p>,
        <p>
          Population:
          2000000000
</p>,
        <ul>
          Residents:
          <li>
                    Leia Organa
          </li>
          <li>
                    Bail Prestor Organa
          </li>
          <li>
                    Raymus Antilles
          </li>
</ul>,
        <button
          className="favorite-button"
          onClick={[Function]}
>
          Favorite
</button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Climate: ",
            "temperate",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Climate: ",
          "temperate",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Terrain: ",
            "grasslands, mountains",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Terrain: ",
          "grasslands, mountains",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population:",
            "2000000000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Population:",
          "2000000000",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Residents:",
            Array [
              <li>
                Leia Organa
</li>,
              <li>
                Bail Prestor Organa
</li>,
              <li>
                Raymus Antilles
</li>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Residents:",
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Leia Organa",
            },
            "ref": null,
            "rendered": "Leia Organa",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Bail Prestor Organa",
            },
            "ref": null,
            "rendered": "Bail Prestor Organa",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "Raymus Antilles",
            },
            "ref": null,
            "rendered": "Raymus Antilles",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "favorite-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5 />,
          <p>
            Climate: 
            temperate
             
</p>,
          <p>
            Terrain: 
            grasslands, mountains
</p>,
          <p>
            Population:
            2000000000
</p>,
          <ul>
            Residents:
            <li>
                        Leia Organa
            </li>
            <li>
                        Bail Prestor Organa
            </li>
            <li>
                        Raymus Antilles
            </li>
</ul>,
          <button
            className="favorite-button"
            onClick={[Function]}
>
            Favorite
</button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Climate: ",
              "temperate",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Climate: ",
            "temperate",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Terrain: ",
              "grasslands, mountains",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Terrain: ",
            "grasslands, mountains",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population:",
              "2000000000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population:",
            "2000000000",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Residents:",
              Array [
                <li>
                  Leia Organa
</li>,
                <li>
                  Bail Prestor Organa
</li>,
                <li>
                  Raymus Antilles
</li>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Residents:",
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "Leia Organa",
              },
              "ref": null,
              "rendered": "Leia Organa",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Bail Prestor Organa",
              },
              "ref": null,
              "rendered": "Bail Prestor Organa",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "Raymus Antilles",
              },
              "ref": null,
              "rendered": "Raymus Antilles",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "favorite-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CARD TESTS matches the snapshot when a vehicle is passed down 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    checkForFaves={[Function]}
    info={
        Object {
            "class": "wheeled",
            "id": 0,
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": "30",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Sand Crawler
</h5>,
        <p>
          Model: 
          Digger Crawler
           
</p>,
        <p>
          Class: 
          wheeled
</p>,
        <p>
          Passengers:
          30
</p>,
        <button
          className="favorite-button"
          onClick={[Function]}
>
          Favorite
</button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sand Crawler",
        },
        "ref": null,
        "rendered": "Sand Crawler",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Model: ",
            "Digger Crawler",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Model: ",
          "Digger Crawler",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Class: ",
            "wheeled",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Class: ",
          "wheeled",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Passengers:",
            "30",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Passengers:",
          "30",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "favorite-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Sand Crawler
</h5>,
          <p>
            Model: 
            Digger Crawler
             
</p>,
          <p>
            Class: 
            wheeled
</p>,
          <p>
            Passengers:
            30
</p>,
          <button
            className="favorite-button"
            onClick={[Function]}
>
            Favorite
</button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sand Crawler",
          },
          "ref": null,
          "rendered": "Sand Crawler",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Model: ",
              "Digger Crawler",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Model: ",
            "Digger Crawler",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Class: ",
              "wheeled",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Class: ",
            "wheeled",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passengers:",
              "30",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Passengers:",
            "30",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "favorite-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
